src/core/services/VaultService.ts

```diff
@@ -256,4 +260,30 @@ export class VaultServiceImpl implements VaultService {
         });
     }
   }
+
+  public async approveDeposit(props: ApproveDepositProps): Promise<Boolean | TransactionResponse> {
+    const { network, tokenAddress, amount, accountAddress, vaultAddress } = props;
+    const yearn = this.yearnSdk.getInstanceOf(network);
+
+    return yearn.vaults.approveDeposit(accountAddress, vaultAddress, tokenAddress, amount);
+  }
+
+  public async approveZapOut(props: ApproveZapOutProps): Promise<Boolean | TransactionResponse> {
+    const { network, vault, accountAddress } = props;
+    const yearn = this.yearnSdk.getInstanceOf(network);
+
+    return yearn.vaults.approveWithdraw(vault.address, vault.token, accountAddress);
```

Comment:
the Token argument should be the intended target token to withdraw to. Its not always the underlying token (vault.token). Would suggest we just receive directly the addresses instead of the whole Vault data structure.

Reply:
fixed on 227dce8ccc7427637245d1ffbe9f21f0819cece8